controllers/discovery/initDiscoveryUserData.js

```diff
@@ -69,11 +75,23 @@ const InitDiscoveryUserData = async (req, res) => {
     });
     let suggestedUsers = usersWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block executes a SQL query to get the total number of users. It's good that you're using parameterized queries (`:userId`) to prevent SQL injection. However, consider moving this database logic to a separate service layer to improve modularity and separation of concerns.
```diff
-    let totalData = await sequelize.query(totalDataQuery, {
-      type: QueryTypes.SELECT,
-      replacements: {
-        userId
-      },
-      raw: true
-    });
+    let totalData = await userService.getTotalUsers(userId);
```
In the userService:
```javascript
async function getTotalUsers(userId) {
  const totalDataQuery = `
    SELECT 
        count(A.user_id) as total_data
    FROM users A
    WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false`;
  const totalData = await sequelize.query(totalDataQuery, {
    type: QueryTypes.SELECT,
    replacements: {
      userId
    },
    raw: true
  });
  return totalData?.[0]?.total_data || 0;
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->